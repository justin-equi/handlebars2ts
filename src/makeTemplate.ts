import { textToIntf } from "./textToIntf";

export function makeTemplate(template: string, name: string = '__default') {
	return `
import {compile} from 'handlebars';

export const ${name} = compile<${textToIntf(template)}>(${JSON.stringify(
		template,
	)});
export default ${name};
  `;
}
