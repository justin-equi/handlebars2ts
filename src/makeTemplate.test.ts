import { expect, it } from "vitest";
import { makeTemplate } from "./makeTemplate";
import { CompilerOptions, ModuleKind, transpileModule } from "typescript";
import vm from "node:vm";
import type { Context } from "node:vm";

function transpile(
	source: string,
	compilerOptions: CompilerOptions = { module: ModuleKind.CommonJS },
) {
	return transpileModule(source, { compilerOptions }).outputText;
}
it("should make a nice template", () => {
	expect(makeTemplate(`hello, {{{name}}}!`)).toMatchSnapshot();
});

it("should compile", function () {
	const result = new vm.Script(
		transpile(makeTemplate( `hello, {{{name}}}!`)),
		{
			filename: "sampletest.mjs",
		},
	).runInNewContext({
		ModuleKind: ModuleKind.CommonJS,
		displayErrors: true,
		require,
		console,
		import: {},
		module: {},
		exports: {},
	});
	expect(result({ name: "Bob" })).toEqual("hello, Bob!");
});
