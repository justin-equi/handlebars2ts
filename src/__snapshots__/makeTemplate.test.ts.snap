// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should make a nice template 1`] = `
"
import {compile} from 'handlebars';

export const __default = compile<{
	name: string;
}>(\\"hello, {{{name}}}!\\");
export default __default;
  "
`;
